app/controllers/locations/map_controller.rb

```diff
@@ -0,0 +1,27 @@
+require_dependency "locations/application_controller"
+
+module Locations
+  class MapController < ApplicationController
+
+    def index
+        @map = Map.new(id: map_params[:id],loc: map_params[:loc])
+
+        if !@map.url.nil?
+          if @map.url.include? "http"
+            redirect_to @map.url
+          else
+            render plain: "#{@map.url}", status: 200
```

Comment:
When you say "reasonably formatted page," would you like me to mimic the look and feel of the PUL website?  Orange Light (where they are likely to be coming from)?  Do we want to provide any navigation to any other resources from that page?  

If the library codes are the same between bibdata and Drupal (which they appear to be) we can give them a direct link to the library's course reserves page that has further instructions, directions, and contact info.  For example, if it's Lewis, then we redirect here: http://library.princeton.edu/lewis/services/course-reserves